src/components/AuthComponents/RouteGuard.tsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState } from "react";
+import SignUpModal from "@/components/AuthComponents/SignUpModal";
+import LoginModal from "@/components/AuthComponents/LoginModal";
+import axios, { AxiosError } from "axios";
+import { useTranslation } from "react-i18next";
+import useAlertStore from "@/stores/alertStore";
+
+const RouteGuard = ({ children }: Readonly<{ children: React.ReactNode }>) => {
+  const [isAuth, setAuth] = useState(false);
+  const [isSignup, setIsSignup] = useState(false);
+  const { showAlert } = useAlertStore();
+  const { t } = useTranslation("authentification");
+
+  const handleLogin = async (username: string, password: string) => {
+    try {
+      const res = await axios.post("/api/login", { username: username }, {
+        headers: { Authorization: "Basic " + btoa(username + ":" + password) }
+      })
+      if (res.status >= 200 && res.status < 300) {
+        document.cookie = "token=" + btoa(username) + "; SameSite=Strict;";
+        setAuth(true);
+        return "";
+      }
+    } catch(err){
+      if(err instanceof AxiosError){
+        if(err.status==404){
+          showAlert(t("Errors.notFound"), "error");
+          return t("Errors.notFound");
+        }if(err.status==500){
+          showAlert(err.response?.data.error, "error");
+          return err.response?.data.error;
+        }if(err.status==401){
+          return t("Errors.unauthorized");
+        }
+      }
+    }
+    return t("Errors.unknown");
+  };
+
+  const handleSignup = async (
+    username: string,
+    password: string,
+    confirm: string,
+  ) => {
+    if(password!=confirm){
+      return t("Errors.passwordMatch")
+    }
+    axios.post("/api/signup",{username:username}, {
+      headers:{ Authorization:  "Basic " + btoa(username + ":" + password)}
+    }).then(res=>{
+      if(res.status>=200 && res.status<300){
+        document.cookie = "token=" + btoa(username) + "; SameSite=Strict;";
+        setAuth(true);
+        return "";
+      }
+    }).catch(err=>{
+      if(err.response){
+        if(err.response.status==404){
+          showAlert(t("Errors.notFound"), "error");
+          return t("Errors.notFound");
+        }if(err.response.status==409){
+          return t("Errors.usernameTaken");
+        }
+        return err.response.data.error;
+      }
+    })
```

Comment:
I think `return axios...`. The `return "";` would then not be necessary if I'm not mistaken.